File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        queue=deque()
        ans=[]
        if not root:
            return []
        queue.append(root)
        while queue:
            n=len(queue)
            for i in range(n):
                aux=queue.popleft()
                if aux.left:
                    queue.append(aux.left)
                if aux.right:
                    queue.append(aux.right)
                if i==n-1:
                    ans.append(aux.val)

        return ans            
        

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int>res;
        queue<TreeNode*> q;
        q.push(root);
        if(root==NULL){
            return res;
        }
        while(!q.empty()){
            int n=q.size();
            for(int i=0;i<n;i++){
                auto aux=q.front();
                q.pop();
                if(aux->left){
                    q.push(aux->left);
                }
                if(aux->right){
                    q.push(aux->right);
                }
                if(i==n-1){
                    res.push_back(aux->val);
                }
                
            }
        }
        return res;
        
        
    }
};